sklearn/neighbors/tests/test_kd_tree.py

```diff
@@ -1,30 +1,97 @@
 import numpy as np
 import pytest
+from numpy.testing import assert_allclose, assert_equal
 
-from sklearn.neighbors._kd_tree import KDTree
+from sklearn.neighbors._kd_tree import KDTree, KDTree32, KDTree64
+from sklearn.neighbors.tests.test_ball_tree import get_dataset_for_binary_tree
 from sklearn.utils.parallel import Parallel, delayed
 
 DIMENSION = 3
 
 METRICS = {"euclidean": {}, "manhattan": {}, "chebyshev": {}, "minkowski": dict(p=3)}
 
+KD_TREE_CLASSES = [
+    KDTree64,
+    KDTree32,
+    KDTree,
```

Comment:
With my suggestion about setting `KDTree = KDTree64`, they will be the same. I think we can have a simple test to assert they are the same:

```python
def test_KDTree_same_as_KDTree64():
    assert KDTree == KDTree64
```

and remove `KDTree` from `KD_TREE_CLASSES` to reduce testing parameters.

Reply:
If we keep the subclass then:

```python
def test_KDTree_is_KDTree64_subclass():
    assert issubclass(KDTree, KDTree64)
```